### ISSUE TITLE
SMERF missing URL for teachers download

### ISSUE THREAD

**Issue by samhodge-aiml** [google-research/google-research#2095]
The documents say

2. Download teacher checkpoints. Unzip their contents like so,

    ```
    teachers/
      bicycle/
        checkpoint_50000/   # Model checkpoint
        config.gin          # Gin config
      ...
    ```

but it is unclear where this model checkpoint comes from.

**Comment by samhodge-aiml** [comment]
Evidently that is where you put he `camp_zipnerf` checkpoint data.

**Comment by DuVogel87** [comment]
Hello,

Would you please help me? I am trying to install SMERF, but unfortunately, the instructions from GitHub are not working for me. When I try to clone the repository, I get this error:

C:\SMERF>git clone https://github.com/smerf-3d/smerf.git
Cloning into 'smerf'...
info: please complete authentication in your browser...
remote: Repository not found.
fatal: repository 'https://github.com/smerf-3d/smerf.git/' not found

Unfortunately, logging into my GitHub account in my browser seems to have no effect.

Could you please share how you installed it on your PC?

Thanks.

**Comment by samhodge-aiml** [comment]
smerf-3d git checkout is not needed you already have the code from google-research repo in the `smerf` folder

**Comment by DuVogel87** [comment]
Thanks for your fast answer. 

I dont get it. 

If I try to git clone the repo like this:

git clone https://github.com/google-research/google-research/tree/master/smerf/smerf

it is not working! 

I am really not understanding how to clone Smerf to my local drive. Can you give further instructions on this issue please? 

![smerf git clone](https://github.com/google-research/google-research/assets/123517103/24df3ddf-3489-42ce-ab57-0ba8b3ce36d8)

**Comment by samhodge** [comment]
Try this 

Clone a Specific Folder from a GitHub Repository https://medium.com/@gabrielcruz_68416/clone-a-specific-folder-from-a-github-repository-f8949e7a02b4

**Comment by DuVogel87** [comment]
Thanks. But immediatly the first step is not possible because there is no clone URL to copy.

**Comment by DuVogel87** [comment]
But anyways, with your help I figured out, what to look for. I found a nice work around and used this site to download all files:

https://download-directory.github.io/

**Comment by samhodge** [comment]
Good luck, let me know how you get on with the computational resources required to run this project

**Comment by DuVogel87** [comment]
> The documents say
> 
> 2. Download teacher checkpoints. Unzip their contents like so,
> 
>     ```
>     teachers/
>       bicycle/
>         checkpoint_50000/   # Model checkpoint
>         config.gin          # Gin config
>       ...
>     ```
> 
> but it is unclear where this model checkpoint comes from.

Where did you download the teacher checkpoints?

**Comment by samhodge-aiml** [comment]
I copied them from the CamP ZipNeRF model where I trained the radiance field using that project's code base and runtime environment with my own data.

**Comment by duckworthd** [comment]
SMERF author here. Teacher checkpoints are still awaiting legal approval for release. In the meantime, I recommend training models using the `camp_zipnerf` codebase.

**Comment by samhodge** [comment]
Thanks @duckworthd one step ahead of you there. Awaiting computational outcome of the SMERF training

**Comment by samhodge-aiml** [comment]
@duckworthd I am not sure whom is repsonsible for the training code but there is a silly JAX thing where `map.tree` becomes `map_tree` if you are super keen I can do it as a PR.

but it is work you knowing about it because the current code doesn't work with the `requirements.txt` frozen version of JAX

similarly there is a Python versioning thing with cuda versions that says JAX can use cudnn `8.9+` for `cuda_12`, which causes a lack of GPU acceleration unless you are on the ball and clamp the cudnn version to the cudnn version for `cuda_12`

This may be invisible to you if you are all 100% accelerators of a different class, but for the CUDA users over here in the rest of the world it is a bit of a stumbling block.

**Comment by duckworthd** [comment]
Thanks for the heads-up, @samhodge-aiml. To be clear, you're saying that `jax.map.tree` must be replaced with `jax.tree_util.tree_map` to function with the pinned version of JAX?  And that the CUDNN version needs to be pinned to 8.9?

**Comment by samhodge-aiml** [comment]
> Thanks for the heads-up, @samhodge-aiml. To be clear, you're saying that `jax.map.tree` must be replaced with `jax.tree_util.tree_map` to function with the pinned version of JAX? And that the CUDNN version needs to be pinned to 8.9?

I think it is just `jax.tree.map` to `jax.tree_map` worked for me, I hope, but if your syntax is more correct, then by all means do that.

I think it is sort of two layers of indirection, the cudnn for cuda 12.3 needs to be pinned to cudnn 8.9

I think the constraint is only for cuda 12 not cuda 12.4 so it grabs a dependancy for cudnn for 12.4 which is cudnn 9.1 but then that version of cudnn is linked against cuda 12.4 and so the symbols cannot load and as a result jax still works but falls back to no GPU acceleration, so unless you have your eyes on the prize, your CPUs will be hammered and your GPU will be idle, if you want things to be set and forget you really need a test to see that there is some sort of acceleration in place before you start training.
it was basically a future incompatibility that could have not been predicted at the time

**Comment by duckworthd** [comment]
References to `jax.tree.map` have been changed to `jax.tree_util.tree_map`. I also added information in the instructions indicating that this code was found working with CUDA 12.3 and cuDNN 8.9, but do not enforce that in any way.

**Comment by samhodge** [comment]
Notes are good

**Comment by samhodge** [comment]
Because you need to be more explicit 

jaxlib[cuda_12]==0.4.23+cuda12.cudnn89

Also refer to

https://pypi.org/project/nvidia-cudnn-cu12/

https://pypi.org/project/nvidia-cudnn-cu12/8.9.7.29/

https://anaconda.org/nvidia/cuda-toolkit

conda install nvidia/label/cuda-12.3.2::cuda-toolkit

**Comment by samhodge** [comment]
Which Python version did you use?

I think 3.8 or 3.9 are needed and it indicates you are using 3.7

**Comment by samhodge** [comment]
The instructions are incorrect that is why I told you what to do

**Comment by samhodge** [comment]
See

https://github.com/google-research/google-research/issues/2095#issuecomment-2148053328

**Comment by samhodge** [comment]
![image](https://github.com/google-research/google-research/assets/86946/b7969233-1532-4746-afe9-9733030b7a3f)

**Comment by samhodge** [comment]
I have run it on Ubuntu 22.04 LTS

You can try whatever you like

**Comment by samhodge** [comment]
I cannot support you anymore, you will have to understand the instructions given CUDA 12.3 CuDNN 8.9 for CUDA 12.3 and then JAX that supports those CUDA and CUDNN versions 

Via conda and python package index

**Comment by samhodge-aiml** [comment]
Somehow common sense might prevail and you visit

https://storage.googleapis.com/jax-releases/jax_cuda_releases.html And notice that all the wheels are for Linux

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by samhodge** [comment]
https://storage.googleapis.com/jax-releases/jax_cuda_releases.html

There is no CUDA support for JAX on Windows 

![image](https://github.com/google-research/google-research/assets/86946/cf0d5632-8307-4ed9-a58a-e50c9e30e5cd)
